elastic/kibana
# [maps] fix map application is broken if you open a map with layers or sources that do not exist

Fixes https://github.com/elastic/kibana/issues/176318

PR adds try/catch blocks around all usages of `createInstanceError`.

### Test steps
1) start kibana with `yarn start --run-examples`
2) create new map, add new layer `Weather data provided by NOAA`
3) save map
4) re-start kibana with `yarn start`
5) open map saved in step 3. Map should open and layer shoould display error in legend
<img width="500" alt="Screenshot 2024-02-07 at 8 13 01 PM" src="https://github.com/elastic/kibana/assets/373691/a70e7ee9-fe30-499a-add5-8686fb591ce6">
